tests/stockfish/test_models.py

```diff
@@ -271,62 +271,80 @@ def test_set_elo_rating(self, stockfish):
 
-        assert stockfish.get_parameters()["UCI_Elo"] == 2850
+        assert stockfish.get_engine_parameters()["UCI_Elo"] == 2850
 
     def test_specific_params(self, stockfish):
         old_parameters = {
             "Debug Log File": "",
             "Contempt": 0,
             "Min Split Depth": 0,
             "Threads": 1,
-            "Ponder": "false",
+            "Ponder": False,
             "Hash": 16,
             "MultiPV": 1,
             "Skill Level": 20,
             "Move Overhead": 10,
             "Minimum Thinking Time": 20,
             "Slow Mover": 100,
-            "UCI_Chess960": "false",
-            "UCI_LimitStrength": "false",
+            "UCI_Chess960": False,
+            "UCI_LimitStrength": False,
             "UCI_Elo": 1350,
         }
         expected_parameters = old_parameters.copy()
         stockfish.set_skill_level(1)
         expected_parameters["Skill Level"] = 1
-        assert stockfish.get_parameters() == expected_parameters
+        assert stockfish.get_engine_parameters() == expected_parameters
         assert stockfish._DEFAULT_STOCKFISH_PARAMS == old_parameters
         stockfish.set_skill_level(20)
         expected_parameters["Skill Level"] = 20
-        assert stockfish.get_parameters() == old_parameters
+        assert stockfish.get_engine_parameters() == old_parameters
         assert stockfish._DEFAULT_STOCKFISH_PARAMS == old_parameters
 
         stockfish.update_engine_parameters({"Threads": 4})
         expected_parameters["Threads"] = 4
-        assert stockfish.get_parameters() == expected_parameters
+        assert stockfish.get_engine_parameters() == expected_parameters
         stockfish.update_engine_parameters({"Hash": 128})
         expected_parameters["Hash"] = 128
-        assert stockfish.get_parameters() == expected_parameters
+        assert stockfish.get_engine_parameters() == expected_parameters
         stockfish.update_engine_parameters({"Hash": 256, "Threads": 3})
         expected_parameters.update({"Hash": 256, "Threads": 3})
-        assert stockfish.get_parameters() == expected_parameters
+        assert stockfish.get_engine_parameters() == expected_parameters
+
+    def test_update_engine_parameters_wrong_type(self, stockfish):
+        with pytest.raises(ValueError):
+            stockfish.update_engine_parameters({"Ponder": "true"})
+        with pytest.raises(ValueError):
+            stockfish.update_engine_parameters({"Ponder": "false"})
+        with pytest.raises(ValueError):
+            stockfish.update_engine_parameters({"UCI_LimitStrength": "true"})
+        with pytest.raises(ValueError):
+            stockfish.update_engine_parameters({"UCI_LimitStrength": "false"})
+        with pytest.raises(ValueError):
+            stockfish.update_engine_parameters({"UCI_Chess960": "true"})
+        with pytest.raises(ValueError):
+            stockfish.update_engine_parameters({"UCI_Chess960": "false"})
```

Comment:
```suggestion
    @pytest.mark.parametrize(
        "parameters",
        [
            {"Ponder": "true"},
            {"Ponder": "false"},
            {"UCI_LimitStrength": "true"},
            {"UCI_LimitStrength": "false"},
            {"UCI_Chess960": "true"},
            {"UCI_Chess960": "false"},
        ],
    )
    def test_update_engine_parameters_wrong_type(self, stockfish, parameters):
        with pytest.raises(ValueError):
            stockfish.update_engine_parameters(parameters)
```
Personally, I would prefer it to separate the parameter possibilities because the different cases don't really have anything to do with each other. And if only some tests don't work, it is easier to see where the problem comes from

Reply:
@kieferro Sure, makes sense - I'll update the test to that.